lib/src/game/game.dart

```diff
@@ -30,8 +30,24 @@ abstract class Game {
   /// Currently attached build context. Can be null if not attached.
   BuildContext get buildContext => _gameRenderBox?.buildContext;
 
+  /// Current size of the game as provided by the framework; it will be null if not attached.
+  ///
+  /// Use [size] and [hasLayout] for safe access.
+  Vector2 _size;
+
+  /// Indicates if the this game instance had its layout layed into the GameWidget
+  /// Only this is true, the game is ready to have its size used or in the case
+  /// of a BaseGame, to receive components.
+  bool get hasLayout => size != null;
```

Comment:
this got me thinking, is this the same thing as isAttached?

rn we wait for onResize which comes from the build method builder so I believe it has the actual size from `constraints.biggest`.

we could read `_gameRenderBox.size` on the onAttach method instead and then isAttached and hasLayout would be the same. however I don't think we can do that. something to study for a potential followup @erickzanardo

Reply:
Using isAttached was the first thing I tried, but it don't work as the onLoad of the game happens before the the attach method is called on game. Don't think there is much alternative to this

Reply:
Btw I noticed that you committed on the PR, this line probably should use _size instead of size

Reply:
yep just fixed that, nice catch! took me longer than you did to find the issue

Reply:
would be nice to have in our docs a diagram that shows the steps relating flutter steps with flame steps, like

attach -> layout -> onLoad -> game loop

and what happens/is available on each step